### ISSUE TITLE
Error in lists plugin when <ol> directly includes text node

### ISSUE THREAD

**Issue by dstillman** [tinymce/tinymce#5548]
**What is the current behavior? Describe the bug**

Pressing Tab in a list that contains a text node directly within an `<ol>` (instead of within an `<li>`) throws an error.

**Please provide the steps to reproduce**

Load 5.2.1 in [fiddle.tinymce.com](http://fiddle.tinymce.com) and set this as the source of the note:

```html
<ol>
	<li>x
		<ol>
			<li>PRESS TAB ON THIS LINE</li>
		</ol>
	</li>
	<li>x
		<ol>
			<li>x
				<ol>
					<li>x</li>
					OUTSIDE OF LI
				</ol>
			</li>
		</ol>
	</li>
</ol>
```

Press Tab on the designated line and `xs is undefined` will be thrown in the error console from this function:

```javascript
    var each = function (xs, f) {
      for (var i = 0, len = xs.length; i < len; i++) {
        var x = xs[i];
        f(x, i);
      }
    };
```

The problem, I'm guessing, is that `dom().attributes` is called on the text node:

```javascript
    var clone = function (element) {
      return foldl(element.dom().attributes, function (acc, attr) {
        acc[attr.name] = attr.value;
        return acc;
      }, {});
    };
```

If I change that to `element.dom().attributes || []`, the text node is removed and an empty list item takes its place, which seems like an acceptable result, though I guess it might be better if the text node were converted to a list item.

**Comment by ltrouton** [comment]
As per the [OL element MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ol), including a text node in a `<ol>` or `<ul>` tag is invalid HTML. As a result, the best course of action is to remove the invalid HTML manually or fix the HTML from its source. 

In saying that, we will investigate possible ways to prevent this exception but make no guarantees since the HTML at its core is invalid.

**Comment by dstillman** [comment]
Yeah, I understand it's invalid HTML. But all major browsers do render the line, so dealing with it gracefully — and ideally without throwing away the content — seems ideal. If someone is just copying formatted text from somewhere else, debugging the HTML might not be a realistic option, and they wouldn't expect a visible line to be dropped on paste or Tab.

**Comment by EugeneHerasymchuk** [comment]
There should be a way to disable "formatting" of the pasted HTML (for example via View -> Source Code plugin ), I agree.
And I'm sure there is somewhere a way, but struggling to make it work as well

**Comment by huineng** [comment]
i'm seeing this problem also , why not simply add a check 

```
var each = function (xs, f) {
      if ! xs return;
      for (var i = 0, len = xs.length; i < len; i++) {
        var x = xs[i];
        f(x, i);
      }
    };
```

**Comment by jordanade** [comment]
I'm having this issue as well in TinyMCE 5.

**Comment by TheSpyder** [comment]
This was fixed in TinyMCE 6.4. Apologies that we didn't close this issue at the time.
